src/rule_definition_tools.jl

```diff
@@ -135,16 +143,37 @@ function _normalize_scalarrules_macro_input(call, maybe_setup, partials)
     # For consistency in code that follows we make all partials tuple expressions
     partials = map(partials) do partial
         if Meta.isexpr(partial, :tuple)
-            partial
+            Expr(:tuple, map(esc, partial.args)...)
         else
             length(inputs) == 1 || error("Invalid use of `@scalar_rule`")
-            Expr(:tuple, partial)
+            Expr(:tuple, esc(partial))
         end
     end
 
     return call, setup_stmts, inputs, partials
 end
 
+"""
+    derivatives_given_output(Ω, f, xs...)
```

Comment:
Maybe one could follow the Julia convention and make function `f` the first argument? I don't think it matters but since we can define it in whatever way, why not :shrug:

Reply:
Just to explain the rationale, I was afraid that `derivatives_given_output(f,  Ω, xs...)` would make it less clear that `Ω` is not one of the inputs. I'm loosely following `frule(tangents, f, inputs...)`.